shared/chat/conversation/messages/text/reply.tsx

```diff
@@ -84,58 +107,57 @@ const Reply = (p: {isParentHighlighted: boolean}) => {
         <Kb.Box2 direction="horizontal" style={styles.quoteContainer} />
         <Kb.Box2 direction="vertical" gap="xtiny" style={styles.replyContentContainer}>
           <Kb.Box2 direction="horizontal" fullWidth={true}>
-            <Kb.Box2 direction="horizontal" gap="xtiny" fullWidth={true}>
-              <Kb.Avatar username={username} size={16} />
-              <Kb.Text
-                type="BodySmallBold"
-                style={Styles.collapseStyles([
-                  styles.replyUsername,
-                  isParentHighlighted && styles.replyUsernameHighlighted,
-                ])}
-              >
-                {username}
-              </Kb.Text>
-            </Kb.Box2>
+            <AvatarHolder isParentHighlighted={isParentHighlighted} />
           </Kb.Box2>
           <Kb.Box2 direction="horizontal" fullWidth={true} gap="tiny">
-            {!!imageURL && (
-              <Kb.Box2 direction="vertical" style={styles.replyImageContainer}>
-                <Kb.Box style={{...(sizing ? sizing.margins : {})}}>
-                  <Kb.Image
-                    src={imageURL}
-                    onLoad={() => setImageLoaded(true)}
-                    style={{...(sizing ? sizing.dims : {})}}
-                  />
-                  {!imageLoaded && <Kb.ProgressIndicator style={styles.replyProgress} />}
-                </Kb.Box>
-              </Kb.Box2>
-            )}
+            {showImage && <ReplyImage />}
             <Kb.Box2 direction="horizontal" style={styles.replyTextContainer}>
-              {!deleted ? (
-                <Kb.Text
-                  type="BodySmall"
-                  style={Styles.collapseStyles([isParentHighlighted && styles.textHighlighted])}
-                  lineClamp={3}
-                >
-                  {text}
-                </Kb.Text>
-              ) : (
+              {isDeleted ? (
                 <Kb.Text type="BodyTiny" style={styles.replyEdited}>
                   The original message was deleted.
                 </Kb.Text>
+              ) : (
+                <ReplyText isParentHighlighted={isParentHighlighted} />
               )}
             </Kb.Box2>
           </Kb.Box2>
-          {edited && (
+          {showEdited && (
             <Kb.Text type="BodyTiny" style={styles.replyEdited}>
               EDITED
             </Kb.Text>
           )}
         </Kb.Box2>
       </Kb.Box2>
-    </Kb.ClickableBox>
+    </Kb.ClickableBox2>
   )
-}
+})
+
+const Reply = React.memo(function Repy(p: {isParentHighlighted: boolean}) {
```

Comment:
this is the general patterns to skip time killing 'mid' components. We have the connector, a memo using bools to hold the structure. then context connected deep children which mutate and move the thrash to the bottom of the stack

Reply:
<img width="125" alt="mid memo" src="https://user-images.githubusercontent.com/516435/212439713-aa4462c4-f864-4186-a547-309ace7ecb8d.png">